workitem_blackbox_test.go

```diff
@@ -130,3 +137,162 @@ func TestListByFields(t *testing.T) {
 
 	test.DeleteWorkitemOK(t, nil, nil, controller, wi.ID)
 }
+
+func getExpiredAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	token.Claims = jwt.MapClaims{"exp": float64(time.Now().Unix() - 100)}
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Bearer %s", tokenStr)
+}
+
+func getMalformedAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Malformed Bearer %s", tokenStr)
+}
+
+func getValidAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Bearer %s", tokenStr)
+}
+
+func TestUnauthorizeCreateWI(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	privatekey, err := jwt.ParseRSAPrivateKeyFromPEM(([]byte(RSAPrivateKeyTest)))
```

Comment:
yeah, i have checked that PR, and will modify accordingly. thanks for making methods public :)